### ISSUE TITLE
lru_crawler metadump misses last entry

### ISSUE THREAD

**Issue by calin-iorgulescu** [memcached/memcached#362]
Steps to reproduce:

1. Load some random records (10 records here):
```
add usertable-user6284781860667377211 0 2147483647 1149
STORED
add usertable-user8517097267634966620 0 2147483647 1143
STORED
add usertable-user1820151046732198393 0 2147483647 1144
STORED
add usertable-user4052466453699787802 0 2147483647 1151
STORED
add usertable-user3232700585171816769 0 2147483647 1146
STORED
add usertable-user1000385178204227360 0 2147483647 1150
STORED
add usertable-user7697331399106995587 0 2147483647 1147
STORED
add usertable-user5465015992139406178 0 2147483647 1144
STORED
add usertable-user6873002678636213555 0 2147483647 1149
STORED
add usertable-user9105318085603802964 0 2147483647 1155
STORED
```
Confirm loading:
```
stats items
STAT items:13:number 10
STAT items:13:number_hot 0
STAT items:13:number_warm 0
STAT items:13:number_cold 10
STAT items:13:age_hot 0
STAT items:13:age_warm 0
STAT items:13:age 161
STAT items:13:evicted 0
STAT items:13:evicted_nonzero 0
STAT items:13:evicted_time 0
STAT items:13:outofmemory 0
STAT items:13:tailrepairs 0
STAT items:13:reclaimed 0
STAT items:13:expired_unfetched 0
STAT items:13:evicted_unfetched 0
STAT items:13:evicted_active 0
STAT items:13:crawler_reclaimed 0
STAT items:13:crawler_items_checked 20
STAT items:13:lrutail_reflocked 0
STAT items:13:moves_to_cold 10
STAT items:13:moves_to_warm 0
STAT items:13:moves_within_lru 0
STAT items:13:direct_reclaims 0
STAT items:13:hits_to_hot 0
STAT items:13:hits_to_warm 0
STAT items:13:hits_to_cold 10
STAT items:13:hits_to_temp 0
END
```
2. Try to use `lru_crawler` to get items (only _9_ are returned, last one is omitted):
```
lru_crawler metadump all
key=usertable-user6284781860667377211 exp=2147483647 la=1521142095 cas=1 fetch=yes cls=13 size=1241
key=usertable-user8517097267634966620 exp=2147483647 la=1521142095 cas=2 fetch=yes cls=13 size=1235
key=usertable-user1820151046732198393 exp=2147483647 la=1521142095 cas=3 fetch=yes cls=13 size=1236
key=usertable-user4052466453699787802 exp=2147483647 la=1521142095 cas=4 fetch=yes cls=13 size=1243
key=usertable-user3232700585171816769 exp=2147483647 la=1521142095 cas=5 fetch=yes cls=13 size=1238
key=usertable-user1000385178204227360 exp=2147483647 la=1521142095 cas=6 fetch=yes cls=13 size=1242
key=usertable-user7697331399106995587 exp=2147483647 la=1521142095 cas=7 fetch=yes cls=13 size=1239
key=usertable-user5465015992139406178 exp=2147483647 la=1521142095 cas=8 fetch=yes cls=13 size=1236
key=usertable-user6873002678636213555 exp=2147483647 la=1521142095 cas=9 fetch=yes cls=13 size=1241
END
```

Records are generated using the Yahoo! Cloud Serving Benchmark ([https://github.com/brianfrankcooper/YCSB](https://github.com/brianfrankcooper/YCSB)).

**Comment by calin-iorgulescu** [comment]
Identified problem to be in https://github.com/memcached/memcached/blob/master/crawler.c#L389, the `remaining` count gets decremented before the comparison, leading to the last entry being overlooked.

Attempted a fix in #363 . Please review.

**Comment by dormando** [comment]
released in 1.5.7
